rust/template/src/flatbuf.rs

```diff
@@ -464,15 +464,38 @@ impl<'a> FromFlatBuffer<fb::__Command<'a>> for DDValueUpdate {
         let val_table = cmd.val().ok_or_else(|| {
             format!("Update::from_flatbuf: invalid buffer: failed to extract value")
         })?;
-        let val = relval_from_flatbuf(relid, val_table)?;
-        match cmd.weight() {
-            1 => Ok(DDValueUpdate(Update::Insert { relid, v: val })),
-            (-1) => Ok(DDValueUpdate(Update::DeleteValue { relid, v: val })),
-            w => Err(format!("Update::from_flatbuf: non-unit weight {}", w)),
+        match cmd.operation() {
+            Operation_InsertOrDelete => {
+                let val = relval_from_flatbuf(relid, val_table)?;
+                match cmd.weight() {
+                    1 => Ok(DDValueUpdate(Update::Insert { relid, v: val })),
+                    (-1) => Ok(DDValueUpdate(Update::DeleteValue { relid, v: val })),
+                    w => Err(format!("Update::from_flatbuf: non-unit weight {}", w)),
+                }
+            }
+            Operation_InsertOrUpdate => {
+                let val = relval_from_flatbuf(relid, val_table)?;
+                Ok(DDValueUpdate(Update::InsertOrUpdate { relid, v: val }))
+            }
+            Operation_DeleteByKey => {
+                let val = relkey_from_flatbuf(relid, val_table)?;
+                Ok(DDValueUpdate(Update::DeleteKey { relid, k: val }))
+            }
+            o => Err(format!(
+                "Update::from_flatbuf: unknown operation code {}",
+                o
+            )),
         }
     }
 }
 
+// These should be ideally imported from flatbuf, but
+// I don't know how to do this.  In the meantime keep in sync
+// with Flatbuffer.hs
+const Operation_InsertOrDelete: i8 = 0;
```

Comment:
I think this should be an enum
```
enum FBOperation {
    InsertOrDelete = 0,
    InsertOrUpdate = 1,
    DeleteByKey = 2,
}
```

You can cast it like so: `FBOperation::InsertOrUpdate as i8`.